text/0813-template-layering.md

```diff
@@ -0,0 +1,984 @@
+   This would have been fine if we chose to go that route in RFC #779 and have
+   plans to evolve the official version to have the same feature. With RFC #779
+   favoring the `<template>` approach, this is not going to happen and this
+   conflict will now be quite confusing, especially since they actually do very
+   different things (e.g. the "official" `hbs` does not return a component).
+
+3. It uses a standard JavaScript syntax in a non-standard way, semantics-wise.
+   It looks like an inert JavaScript string but has access to the lexical scope
+   around it (without using the `${ ... }` syntax), which can be confusing. It
+   also means that it is impossible to provide a runtime implementation using
+   the same syntax that works in environments that does not permit a build step.
+
+4. It uses `static template = ` to associate templates to classes. This implies
+   runtime behavior that is actually not true (e.g. TypeScript will believe the
+   `template` property exists on the class).
+
+5. Values consumed by the template but is otherwise unused in the rest of the
+   file may generate errors/warnings from linters or language servers.
+
+To address these issues, this RFC propose that we introduce a new `template()`
+function that servers as a middle ground between the `<template>` language
+extension and the low-level primitives.
+
+- It is designed to have the same semantics as the `<template>` feature (such
+  as the strict-mode opt-in, returning a template-only component when not
+  attached to a class), providing the same high-level programming model for
+  authoring components in standard JavaScript environments.
+
+- It requires manually supplying the lexical scope variable bindings (the
+  "scope function").
+
+- It uses a [static initializer block][static-block] to associate the template.
+
+  [static-block]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Class_static_initialization_blocks
+
+- While it is more cumbersome to use and have a degraded experience (e.g.
+  possibly lacking hbs syntax highlighting) compared to first-class component
+  templates, it is still designed to be ergonomic to use, to the extent
+  possible in standard JavaScript environments.
+
+- It is designed to be pre-processed at build time where possible, just like
+  the `<template>` feature and today's "official" `hbs` tag. However, in cases
+  where this is not possible, there will be a runtime implementation available
+  provided the template compiler is also available at runtime.
+
+- For environments where it is possible to configure the development and build
+  tools to recognize the format, the tagged template literal form can be used,
+  which automatically captures the lexical scope variable bindings. This
+  directly replaces [ember-template-imports][ember-template-imports]'s "hbs
+  backtick" format but brings its naming into alignment with the `<template>`
+  feature.
```

Comment:
What would be an environment which can handle *that* preprocessing step but which *couldn't* handle `<template>`?